runtime/near-vm-runner/src/wasmer2_runner.rs

```diff
@@ -301,62 +271,245 @@ pub(crate) fn wasmer2_vm_hash() -> u64 {
     WASMER2_CONFIG.config_hash()
 }
 
-pub(crate) fn default_wasmer2_store() -> Store {
-    // We only support singlepass compiler at the moment.
-    assert_eq!(WASMER2_CONFIG.compiler, WasmerCompiler::Singlepass);
-    let compiler = Singlepass::new();
-    // We only support universal engine at the moment.
-    assert_eq!(WASMER2_CONFIG.engine, WasmerEngine::Universal);
-    let engine = wasmer::Universal::new(compiler).features(WASMER_FEATURES).engine();
-    Store::new(&engine)
+#[derive(Clone)]
+pub(crate) struct VMArtifact {
+    artifact: Arc<dyn Artifact>,
```

Comment:
ditto here, I'd probably spell this as `wasmer_engine::Artifact`, just so it's obvious which one of the two artifacts is ours.